cprover_bindings/src/goto_program/typ.rs

```diff
@@ -973,6 +973,19 @@ impl Type {
         Pointer { typ: Box::new(self) }
     }
 
+    /// Convert type to its unsigned counterpart if possible.
+    /// For types that are already unsigned, this will return self.
+    /// Note: This will expand any typedef.
+    pub fn to_unsigned(&self) -> Option<Self> {
+        let concrete = self.unwrap_typedef();
+        match concrete {
+            CInteger(CIntType::SSizeT) => Some(CInteger(CIntType::SizeT)),
```

Comment:
Aren't there some other types that could go here too?

Reply:
I could include `Bool` which would be a no-op and I could handle `CBitField` but I noticed that we don't handle that in neither `signed()` or `unsigned()` so we should probably add some logic there too. For us to add support to `Char` we would need access to the machine model. Is there any other type you would like to see?

Let me know what you prefer and I can add that as a follow up PR.